### DISCUSSION TITLE
Elasticsearch index-prefix date

### DISCUSSION THREAD

**Discussion by trocade** [Security-Onion-Solutions/securityonion#7669] [Unsupported Versions]
So I have been having some issues where a lot of old indexes are being created causing a massive spike in shards for no data.
(10 shards for 1 document)

After some looking into this I realise it is due to this in /opt/so/conf/elasticsearch/ingest/common

```
        "date_index_name": {
                "field": "@timestamp",
                "index_name_prefix": "{{ _index }}-",
                "date_rounding": "d",
                "ignore_failure": true,
                "index_name_format": "yyyy.MM.dd"
        }

```

Since timestamp is set to event.timestamp and not ingest.timestamp that means that if a log has a incorrect date it will create a new index.

In my opinion and what I am trying to change, the date_index_name should be set to ingest.timestamp since that would always be todays date (unless the sensor has incorrect timestamp).

And yes, fixing the faulty devices are of course also something that have to be done :)

EDIT:
So in our environment I have updated the ingest/pipeline succesfully to this (including the salt template):
```
        "date_index_name": {
                "field": "ingest.timestamp",
                "index_name_prefix": "{{ _index }}-",
                "date_rounding": "d",
                "ignore_failure": true,
                "index_name_format": "yyyy.MM.dd"
        }

```

**Comment by TOoSmOotH** [comment]
What if your pipeline was backed up due to an outage and now you are left with a bloated index since the backlogged events get dropped into the same index? What if you were  needing to go back and load an old index from a few months ago because you had an incident on 11/12/2001 but that index doesn't exist because there was a blip that day? Can you really trust that index as having the correct information in it since there could be data in later indices that actually happened on the 12th? I think this creates more problems than it solves. If you don't want to change the time on it you can write a logstash parser that would swap the dates for that device. I would do that before taking the hammer approach from above. Or like you said... you could just fix the time on the machine in question. That has benefits beyond fixing this problem. You will want your timestamps to line up if you ever have to take a case to court or legal action needs to take place.
